### ISSUE TITLE
Cannot read property 'scrollTop' of null

### ISSUE THREAD

**Issue by malinduDeAlwis90** [zefoy/ngx-perfect-scrollbar#227]
Hi,
My versions are:
"@angular/core": "7.0.2",
"ngx-perfect-scrollbar": "^6.3.1",
 "swiper": "^4.4.1",

In my project I'm using perfect-scrollbar together with swiper. 
Everything works fine when we do swipe the tabs.
But when we touch the screen and hold for a little time and then swipe will gives the following error

![image](https://user-images.githubusercontent.com/22950451/48932177-75413600-ef20-11e8-9fb3-ec624dd06af8.png)

This occurs because at this point the setInterval below is never cleared.

perfect-scrollbar.esm.js - line 1045

```javascript
  function touchEnd() {
    if (i.settings.swipeEasing) {
      clearInterval(easingLoop);
      easingLoop = setInterval(function() {
        if (i.isInitialized) {
          clearInterval(easingLoop);
          return;
        }

        if (!speed.x && !speed.y) {
          clearInterval(easingLoop);
          return;
        }

        if (Math.abs(speed.x) < 0.01 && Math.abs(speed.y) < 0.01) {
          clearInterval(easingLoop);
          return;
        }

        applyTouchMove(speed.x * 30, speed.y * 30);

        speed.x *= 0.8;
        speed.y *= 0.8;
      }, 10);
    }
  }
```
If there's a way to override this method we can fix it.

Any help?
Thanks,
Malindu

**Comment by sconix** [comment]
At least I am not aware of any easy way, maybe some monkey patching. You maybe should also file a bug for perfect-scrollbar.

**Comment by malinduDeAlwis90** [comment]
Hello @sconix I'm not clear with your answer. Could you please clarify?

**Comment by sconix** [comment]
Its a common method for patching existing code, I have never used it, but know that Angular/Typescript has used it. So you need to google for more info and I am not sure if that even helps.

As a side note we also use Swiper and this, but have not seen this error, but I guess our usage might be bit different than yours.

**Comment by malinduDeAlwis90** [comment]
The thing is swipe slides are dynamically created and destroyed on swipe. So sometimes 'scrollTop' is called on a destryed element. Then this problem occurs.

**Comment by sconix** [comment]
Yes and this traditional JS libraries (such as Swiper and PS) have usually some problems with dynamic DOM updating. Usually these can be fixed by carefully planning the changes in the Angular app, but sometimes not without a fix for the actual library. Anyway can not think anything that I could do in this library to fix this so better to file a bug for PS.

**Comment by malinduDeAlwis90** [comment]
Thanks @sconix , can you tell me how to file a bug to PS ?

**Comment by sconix** [comment]
Just make an issue in PS github page. Link is in the Readme.

**Comment by malinduDeAlwis90** [comment]
Thanks @sconix

**Comment by olaf89** [comment]
Im having same errors reported to sentry, it would be awsome to add some nullcheck there

**Comment by sconix** [comment]
You need to ask that in PS issues, I am not the maintainer for PS.
